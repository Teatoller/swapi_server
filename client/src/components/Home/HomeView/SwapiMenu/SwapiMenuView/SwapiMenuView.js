export const SwapiMenuView = ({ searchSpace, handleSearchSpace }) => {
  return (
    <>
      <div className="mb-0">
        <div className="d-inline p-1 bg-light text-white">
          <form class="form-inline my-2 my-lg-0">
            <input
              className="border border-info rounded w-25 text-muted text-center"
              type="text"
              placeholder="Search"
              onChange={(e) => searchSpace(e)}
            />
            <button
              className="border border-info rounded bg-primary m-1"
              onClick={handleSearchSpace}
              type="button"
            >
              <i className="bi bi-search text-white"></i>
            </button>
          </form>
        </div>
      </div>
    </>
  );
};
export default SwapiMenuView;
