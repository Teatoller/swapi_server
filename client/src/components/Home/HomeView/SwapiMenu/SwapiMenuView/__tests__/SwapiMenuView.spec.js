import React from "react";
import { render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import renderer from "react-test-renderer";
import { SwapiMenuView } from "../SwapiMenuView";

describe("SwapiMenuView", () => {
  test('renders SwapimenuView search input', () => {
    render(<SwapiMenuView />);
    expect(screen.getByRole('textbox')).toBeInTheDocument();
  });

  test('renders SwapimenuView button', () => {
    render(<SwapiMenuView />);
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(screen.queryByText(/Searches for Luke/)).toBeNull();
  });

  it("snapshot renders correctly", () => {
    const component = renderer.create(<SwapiMenuView />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
