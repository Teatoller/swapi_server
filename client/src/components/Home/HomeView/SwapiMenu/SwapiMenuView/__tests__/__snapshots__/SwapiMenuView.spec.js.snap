// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapiMenuView snapshot renders correctly 1`] = `
<div
  className="mb-0"
>
  <div
    className="d-inline p-1 bg-light text-white"
  >
    <form
      class="form-inline my-2 my-lg-0"
    >
      <input
        className="border border-info rounded w-25 text-muted text-center"
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
      <button
        className="border border-info rounded bg-primary m-1"
        type="button"
      >
        <i
          className="bi bi-search text-white"
        />
      </button>
    </form>
  </div>
</div>
`;
